src/main.rs

```diff
@@ -906,12 +915,78 @@ where
     // the receiver of the txn is the sender of the signed delegate action
     let receiver_id = &signed_delegate_action.delegate_action.sender_id;
     let da_receiver_id = &signed_delegate_action.delegate_action.receiver_id;
-
     // check that the delegate action receiver_id is in the whitelisted_contracts
     let is_whitelisted_da_receiver = WHITELISTED_CONTRACTS
         .iter()
         .any(|s| s == da_receiver_id.as_str());
-    if !*USE_FASTAUTH_FEATURES && !is_whitelisted_da_receiver {
+    if USE_EXCHANGE.clone() {
+        let non_delegate_actions: Vec<Action> =
+            signed_delegate_action.delegate_action.get_actions();
+        if *USE_WHITELISTED_DELEGATE_ACTION_RECEIVER_IDS {
+                        // check if the delegate action receiver_id (account sender_id) if a whitelisted delegate action receiver
+                        let is_whitelisted_sender = WHITELISTED_DELEGATE_ACTION_RECEIVER_IDS
+                            .iter()
+                            .any(|s| s == receiver_id.as_str());
+                        if !is_whitelisted_sender {
+                            return Err(RelayError {
+                                status_code: StatusCode::BAD_REQUEST,
+                                message: "Delegate Action Sender_id {receiver_id} is not whitelisted".to_string(),
+                            });
+                        }
+        }
```

Comment:
can throw an exception in the else since USE_EXCHANGE implies USE_WHITELISTED_DELEGATE_ACTION_RECEIVER_IDS